content/02-model/02-model-metrics/document.qmd

```diff
@@ -67,82 +67,156 @@ data <- bike_model_data %>%
+suppressPackageStartupMessages(handler_startup(v))
+
+# Specifically load the packages required by the model. Check 
+# `v$metadata$required_pkgs` to see the required pacakges. These packages must
+# be specicially defined so that RStudio Connect knows to install them when
+# deploying this document.
+library(parsnip)
+library(ranger)
+library(recipes)
+library(workflows)
+
+
+preds <- augment(v, test_data)
+
+latest_metrics <- preds %>%
+  arrange(date) %>%
+  vetiver_compute_metrics(
+    date_var = date,
+    period = "day",
+    truth = n_bikes,
+    estimate = .pred
+  )
 
-glimpse(all_time_metrics)
+pin_name <- "sam.edwardes/bike-predict-model-metrics"
+
+if (pin_exists(board, pin_name)) {
+  print("Pin already exists, updating existing pin...")
+  vetiver_pin_metrics(board, latest_metrics, pin_name, overwrite = TRUE)
+} else {
+  print("Creating metrics pin for the first time...")
+  pin_write(board, latest_metrics, pin_name)
+}
 ```
 
-Plot the model performance over time.
+### Overall
 
-```{r all_time_metrics_plot}
-fig <- all_time_metrics %>%
-  tidyr::pivot_longer(cols = c(-data_type, -date), names_to = "metric") %>%
-  ggplot(aes(x = date, y = value, color = data_type)) +
-  geom_line() +
-  geom_point() +
-  facet_wrap(~ metric) +
+The latest evaluation metrics:
+
+```{r view_latest_metrics}
+# Explicitly define packages used for model metrics by vetiver so that Connect 
+# is able to redeploy.
+library(slider)
+
+latest_metrics %>%
+  kableExtra::kable()
+```
+
+### By Prediction
+
+```{r preds_plot}
+preds %>%
+  ggplot(aes(n_bikes, .pred)) +
+  geom_abline(slope = 1, lty = 2, color = "gray60", size = 1.2) +
+  geom_jitter(alpha = 0.5, width = 0.4) +
   labs(
-    title = "Model performance over time",
-    x = "Date",
-    y = "Value"
+    title = glue("Model Performance on Test Data ({test_start_date} to {test_end_date})"),
+    x = "Acutal Number of Bikes",
+    y = "Predicted Number of Bikes"
   )
+```
+
+### By Location
 
-plotly::ggplotly(fig)
+```{r preds_lat_lon_plot}
+preds %>%
+  mutate(.resid = n_bikes - .pred) %>%
+  ggplot(aes(lon, lat, color = .resid)) +
+  geom_point(alpha = 0.8) +
+  scale_color_gradient2() +
+  coord_fixed() + 
+  labs(
+    title = glue("Model Performance by Location on Test Data ({test_start_date} to {test_end_date})"),
+    x = "Longitude",
+    y = "Latitude",
+    color = "Redisual"
+  )
 ```
 
-```{r disconnect_db}
-odbc::dbDisconnect(con)
+:::
+
+## All time Model Metrics
+
+```{r all_time_metrics_plot}
+all_time_metrics <- pin_read(board, pin_name)
+
+vetiver_plot_metrics(all_time_metrics) +
+  labs(
+    title = "All Time Model Metrics",
+    size = "Number of\nObservations"
+  )
 ```
+
+## Ethical considerations
+
+- We considered the ethical impact the use of the model may have. We assessed that the overall risk an incorrect model prediction is relatively low.
+
+## Caveats & recommendations
+
+- This model does guarantee that a bike will be available at a station
```

Comment:
```suggestion
- This model does not guarantee that a bike will be available at a station
```
Is this suggestion correct?

Reply:
Oops... yes I really got that one wrong. Thank you for catching it. Fixed.